8.2.2: pytest is failing in `tests/test_tenacity.py::TestRetryTyping::test_retry_type_annotations` unit

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

Here is pytest output:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-tenacity-8.2.2-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-tenacity-8.2.2-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
============================= test session starts ==============================
platform linux -- Python 3.8.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/tenacity-8.2.2, configfile: setup.cfg
plugins: typeguard-3.0.1
collected 109 items

tests/test_after.py ..                                                   [  1%]
tests/test_asyncio.py ............                                       [ 12%]
tests/test_tenacity.py ................................................. [ 57%]
.......................................F..                               [ 96%]
tests/test_tornado.py ....                                               [100%]

=================================== FAILURES ===================================
_________________ TestRetryTyping.test_retry_type_annotations __________________

self = <tests.test_tenacity.TestRetryTyping testMethod=test_retry_type_annotations>

    @pytest.mark.skipif(sys.version_info < (3, 0), reason="typeguard not supported for python 2")
    def test_retry_type_annotations(self):
        """The decorator should maintain types of decorated functions."""
        # Just in case this is run with unit-test, return early for py2
        if sys.version_info < (3, 0):
            return

        # Function-level import because we can't install this for python 2.
        from typeguard import check_type

        def num_to_str(number):
            # type: (int) -> str
            return str(number)

        # equivalent to a raw @retry decoration
        with_raw = retry(num_to_str)
        with_raw_result = with_raw(1)

        # equivalent to a @retry(...) decoration
        with_constructor = retry()(num_to_str)
        with_constructor_result = with_raw(1)

        # These raise TypeError exceptions if they fail
>       check_type("with_raw", with_raw, typing.Callable[[int], str])
E       TypeError: check_type() takes 2 positional arguments but 3 were given

tests/test_tenacity.py:1545: TypeError
=========================== short test summary info ============================
FAILED tests/test_tenacity.py::TestRetryTyping::test_retry_type_annotations
======================== 1 failed, 108 passed in 3.24s =========================
```
</details>

Here is list of installed modules in build env
<details>

```console
Package            Version
------------------ --------------
attrs              22.2.0
build              0.10.0
distro             1.8.0
exceptiongroup     1.0.0
importlib-metadata 6.1.0
iniconfig          2.0.0
libcomps           0.1.19
packaging          23.0
pip                23.0.1
pluggy             1.0.0
pyproject_hooks    1.0.0
pytest             7.2.2
python-dateutil    2.8.2
rpm                4.17.0
setuptools         65.6.3
setuptools-scm     7.1.0
six                1.16.0
tomli              2.0.1
tornado            6.2
typeguard          3.0.1
typing_extensions  4.5.0
wheel              0.38.4
zipp               3.15.0
```
</details>

--- a/releasenotes/notes/Fix-tests-for-typeguard-3.x-6eebfea546b6207e.yaml
+++ b/releasenotes/notes/Fix-tests-for-typeguard-3.x-6eebfea546b6207e.yaml
@@ -1,0 +1,3 @@
+---
+fixes:
+  - "Fixes test failures with typeguard 3.x"

--- a/tests/test_tenacity.py
+++ b/tests/test_tenacity.py
@@ -1542,10 +1542,10 @@ class TestRetryTyping(unittest.TestCase):
         with_constructor_result = with_raw(1)
 
         # These raise TypeError exceptions if they fail
-        check_type("with_raw", with_raw, typing.Callable[[int], str])
-        check_type("with_raw_result", with_raw_result, str)
-        check_type("with_constructor", with_constructor, typing.Callable[[int], str])
-        check_type("with_constructor_result", with_constructor_result, str)
+        check_type(with_raw, typing.Callable[[int], str])
+        check_type(with_raw_result, str)
+        check_type(with_constructor, typing.Callable[[int], str])
+        check_type(with_constructor_result, str)
 
 
 @contextmanager

